usecases/modules/vectorizer.go

```diff
@@ -245,6 +263,9 @@ func (p *Provider) batchUpdateVector(ctx context.Context, objects []*models.Obje
 		}
 		multiVectors, addProps, vecErrors := vectorizer.VectorizeBatch(ctx, objects, skipRevectorization, cfg)
 		for i := range objects {
+			if ctx.Err() != nil {
+				return nil, fmt.Errorf("context is done: %w", ctx.Err())
+			}
```

Comment:
same, I don't see any expensive happening in this loop that would require a context check for every iteration